src/plugins/dashboard/public/dashboard_container/embeddable/api/run_save_functions.tsx

```diff
@@ -334,3 +334,97 @@ export async function runInteractiveSave(this: DashboardContainer, interactionMo
     })();
   });
 }
+
+export async function duplicate(this: DashboardContainer, redirectTo: DashboardRedirect) {
```

Comment:
It looks like there is a whole bunch of shared code between this new `duplicate` method and `runInteractiveSave` - these two functions are more-or-less doing the same thing, it's just that `runInteractiveSave` has some extra logic for the opening of the save modal. We should be able to reduce this by making a helper function that both of them can use!

As a bonus of doing this, if we ever decide to change how the "duplicate" title logic works, we would only have to change it in one place rather than two - duplicating code like this is always an opportunity for bugs / inconsistent behaviour because it's entirely possible someone will forget to make the changes everywhere needed :)

Reply:
I wonder if the `duplicate` method in this badge should actually just call `runInteractiveSave` directly? IMO there is no reason for a functional difference between duplicating a managed Dashboard vs just duplicating a dashboard...

Reply:
When discussing in https://github.com/elastic/kibana/issues/179152, we decided that the duplicate button in this new button should **not** open the save modal - if we do decide to open the save modal, I think the popover is more-or-less unnecessary and we might as well just open the modal on click of the managed badge (which, in the original discussion, was decided would be pretty obtrusive?) 

Open to changes here, though!

Reply:
Good point, I missed this detail in the issue

> At that point, I think it would be okay to skip the confirmation step unless we decide that renaming the cloned Dashboard is important

I was considering a process where the popover is opened on click with more information, and the `clone` CTA inside the popover opens the existing save as modal.

I understand this could make the process take longer, but I think the consistency outweighs this. I don't feel too strongly either way though.

Reply:
> I understand this could make the process take longer, but I think the consistency outweighs this. I don't feel too strongly either way though. In either case, re-using as much code as possible is best!

I don't have strong feelings either, hahah! :) Maybe we just stick to opening the save modal here for consistency? It would be an easy change to make later if we decide we don't like the two-click. So, just to summarize, we should:

1) Keep the popover with a call to action button for clone
2) Open the save modal on click rather than cloning directly

Reply:
Reminder that this still needs to be addressed 🙇

Reply:
Sorry!!! Just addressed in [06b7ffe](https://github.com/elastic/kibana/pull/189404/commits/06b7ffe577dd930f561428855d2435e1e9f761d2)